src/storage/cache/rpc/packet.rs

```diff
@@ -0,0 +1,500 @@
+            .send(packet.clone())
+            .map_err(|e| {
+                RpcError::InternalError(format!("Failed to send packet to buffer: {e:?}"))
+            })?;
+
+        Ok(())
+    }
+
+    /// Clean pending tasks as timeout
+    pub fn clean_timeout_tasks(&mut self) {
+        let mut packets_inner = self.inner.write();
+        while let Ok(packet) = self.buffer_packets_receiver.try_recv() {
+            let seq = packet.seq();
+            packets_inner.add_task(seq, packet);
+        }
+
+        let current_timestamp = self.current_time.elapsed().as_secs();
+        packets_inner.clean_timeout_tasks(current_timestamp, self.timeout);
+    }
+
+    /// Purge the outdated tasks when connection is timeout
+    pub fn purge_outdated_tasks(&mut self) {
+        let mut packets_inner = self.inner.write();
+        while let Ok(packet) = self.buffer_packets_receiver.try_recv() {
+            let seq = packet.seq();
+            packets_inner.add_task(seq, packet);
+        }
+
+        packets_inner.purge_outdated_tasks();
+    }
+
+    /// Get a task from the packets, and update data in the task
+    pub fn take_task(&self, seq: u64, resp_buffer: &mut BytesMut) -> Result<(), RpcError> {
+        // Try to sync from buffer
+        let mut packets_inner = self.inner.write();
+        while let Ok(packet) = self.buffer_packets_receiver.try_recv() {
+            let seq = packet.seq();
+            packets_inner.add_task(seq, packet);
+        }
+
+        if let Some(packet) = packets_inner.get_task_mut(seq) {
+            let seq = packet.seq();
+            // Update status data
+            // Try to set result code in `set_resp_data`
+            let set_result = packet.set_resp_data(resp_buffer);
+            match set_result {
+                Ok(()) => {
+                    debug!("{:?} Success to set response data, seq: {:?}", self, seq);
+                    packet.set_result(Ok(()));
+                }
+                Err(err) => {
+                    debug!(
+                        "{:?} Failed to set response data: {:?} with error {:?}",
+                        self, seq, err
+                    );
+                    packet.set_result(Err(RpcError::InternalError(format!(
+                        "Failed to set response data: {seq:?} with error {err:?}"
+                    ))));
+                }
+            }
+            packets_inner.remove_task(seq);
+
+            return Ok(());
+        }
+
+        Err(RpcError::InvalidRequest(format!("can not find seq: {seq}")))
+    }
+}
+
+#[cfg(test)]
+#[allow(clippy::unwrap_used)]
+#[allow(dead_code)]
+mod tests {
+    use std::{mem, thread::sleep, time};
+
+    use crate::async_fuse::util::usize_to_u64;
+
+    use super::*;
+
+    #[derive(Debug, Clone)]
+    struct TestRequest {
+        pub mock: u64,
+    }
+
+    impl Encode for TestRequest {
+        fn encode(&self, _buf: &mut BytesMut) {}
+    }
+
+    #[derive(Debug, Clone)]
+    struct TestPacket {
```

Comment:
Instead of Define TestPacket, you can directly use the Packet from message.rs,  to save some coding

Reply:
Ok.